packages/vscode-extension/src/webview/components/Preview.tsx

```diff
@@ -105,11 +107,33 @@ function Preview({
   const openRebuildAlert = useNativeRebuildAlert();
 
   function getTouchPosition(event: MouseEvent<HTMLDivElement>) {
+    const rotation = selectedDeviceSession?.rotation;
+
     const imgRect = previewRef.current!.getBoundingClientRect();
-    const x = (event.clientX - imgRect.left) / imgRect.width;
-    const y = (event.clientY - imgRect.top) / imgRect.height;
-    const clampedX = clamp(x, 0, 1);
-    const clampedY = clamp(y, 0, 1);
+
+    let x = (event.clientX - imgRect.left) / imgRect.width;
+    let y = (event.clientY - imgRect.top) / imgRect.height;
+
+    // Original computed coordinates before rotation
+    let newX = x;
+    let newY = y;
+
+    switch (rotation) {
+      case "LandscapeLeft":
+        newX = 1 - y;
+        newY = x;
+        break;
+      case "LandscapeRight":
+        newX = y;
+        newY = 1 - x;
+        break;
+      default:
+        break;
+    }
+
+    let clampedX = clamp(newX, 0, 1);
+    let clampedY = clamp(newY, 0, 1);
+
```

Comment:
I think this function deserves either a different name (along the lines of `getNormalizedDeviceCoordinates`), or a comment explaining what it does -- or even both.